File: Maintix_API/Models/Historico.cs
using System.ComponentModel.DataAnnotations.Schema;
namespace Maintix_API.Models
{
    public class Historico
    {
        public int Id { get; set; }
        public int EquipoId { get; set; }
        public int? HorasMaquina { get; set; }
        public string? Clase { get; set; }
        public string? Operario { get; set; }
        public string? Incidencias { get; set; }
        public bool Finalizado { get; set; } = false;

        // Fecha del mantenimiento (inicio) y fecha de finalización (no persistidos)
        [NotMapped]
        public DateTime? FechaMantenimiento { get; set; }
        [NotMapped]
        public DateTime? FechaFinalizacion { get; set; }

        // Información referencial que queremos exponer en la API (no persistidos)
        [NotMapped]
        public string? EquipoNumeroSerie { get; set; }
        [NotMapped]
        public string? TipoMantenimientoNombre { get; set; }
    }
}

File: mobile/app/src/main/java/com/tonhete/maintixapp/data/models/Historico.kt
package com.tonhete.maintixapp.data.models

import com.google.gson.annotations.SerializedName

data class Historico(
    @SerializedName("id") val id: Int,
    @SerializedName("equipoId") val equipoId: Int,
    @SerializedName("horasMaquina") val horasMaquina: Int,
    @SerializedName("clase") val clase: String,
    @SerializedName("operario") val operario: String,
    @SerializedName("incidencias") val incidencias: String?,
    @SerializedName("finalizado") val finalizado: Boolean,
    @SerializedName("fechaMantenimiento") val fechaMantenimiento: String,
    @SerializedName("fechaFinalizacion") val fechaFinalizacion: String,
    @SerializedName("equipoNumeroSerie") val equipoNumeroSerie: String,
    @SerializedName("tipoMantenimientoNombre") val tipoMantenimientoNombre: String
)